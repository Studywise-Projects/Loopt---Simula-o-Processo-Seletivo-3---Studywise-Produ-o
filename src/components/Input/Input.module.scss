@import '@/styles/_global.scss';

.input {
  height: 44px;
  width: 320px;

  @media only screen and (min-width: 1024px) {
    width: 420px;
  }
}