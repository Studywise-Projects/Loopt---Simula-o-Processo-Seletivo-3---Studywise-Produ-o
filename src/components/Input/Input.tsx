import { TextField } from '@mui/material';
import styles from './Input.module.scss';
import { ChangeEventHandler } from 'react';

interface IInput {
  required?: boolean;
  id: string;
  label: string;
  type: 'text' | 'password';
  value: string;
  handleChange: ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
}

function Input({
  required = false,
  id,
  label,
  type,
  value,
  handleChange,
}: IInput) {
  return (
    <TextField
      variant='standard'
      required={required}
      id={id}
      label={label}
      type={type}
      value={value}
      onChange={handleChange}
      data-testId='input'
      className={styles.input}
    />
  );
}

export default Input;
