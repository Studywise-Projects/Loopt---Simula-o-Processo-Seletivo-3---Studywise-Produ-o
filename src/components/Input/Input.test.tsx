import { fireEvent, render } from '@testing-library/react';
import Input from './Input';

describe('Input', () => {
  it('Should match the snapshot', () => {
    const { getByLabelText } = render(
      <Input
        id='text-input'
        label='text-label'
        type='text'
        value=''
        handleChange={() => undefined}
        handleKeyPress={undefined}
      />,
    );

    const element = getByLabelText('text-label');
    expect(element).toMatchSnapshot();
  });

  it('Should render correctly', () => {
    const { getByLabelText } = render(
      <Input
        id='text-input'
        label='text-label'
        type='text'
        value=''
        handleChange={() => undefined}
        handleKeyPress={undefined}
      />,
    );

    const element = getByLabelText('text-label');
    expect(element).toBeInTheDocument();
  });

  it('Should render correctly with type = password', () => {
    const { getByLabelText } = render(
      <Input
        id='password-input'
        label='password-label'
        type='password'
        value=''
        handleChange={() => undefined}
        handleKeyPress={undefined}
      />,
    );

    const element = getByLabelText('password-label');
    expect(element).toHaveAttribute('type', 'password');
  });

  it('Should call handleChangeMock function once', () => {
    const handleChangeMock = jest.fn();

    const { getByLabelText } = render(
      <Input
        id='text-input'
        label='text-label'
        type='text'
        value=''
        handleChange={handleChangeMock}
        handleKeyPress={undefined}
      />,
    );

    const element = getByLabelText('text-label');
    fireEvent.change(element, { target: { value: 'input changed' } });
    expect(handleChangeMock).toHaveBeenCalledTimes(1);
  });

  it('Should call handlePressKeyMock function once', () => {
    const handlePressKeyMock = jest.fn();

    const { getByLabelText } = render(
      <Input
        id='text-input'
        label='text-label'
        type='text'
        value=''
        handleChange={() => undefined}
        handleKeyPress={handlePressKeyMock}
      />,
    );

    const element = getByLabelText('text-label');
    fireEvent.keyDown(element, { key: 'Enter', code: 'Enter' });

    expect(handlePressKeyMock).toHaveBeenCalledTimes(1);
  });
});
