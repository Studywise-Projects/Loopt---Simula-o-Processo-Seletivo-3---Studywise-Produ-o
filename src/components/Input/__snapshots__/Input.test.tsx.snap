// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should match the snapshot 1`] = `
<input
  aria-invalid="false"
  class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
  id="text-input"
  type="text"
  value=""
/>
`;
